client/packages/common/src/ui/components/inputs/TextInput/numeric/NumericTextInput.tsx

```diff
@@ -22,7 +38,10 @@ export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
           return;
         }
         const parsed = Number(e.target.value);
-        if (!Number.isNaN(parsed) && !!onChange) onChange(parsed);
+        if (!Number.isNaN(parsed) && !!onChange)
+          onChange(
+            NumUtils.constrain(NumUtils.round(parsed, decimalLimit), min, max)
+          );
       }}
```

Comment:
This is all that is needed to incorporate all the functionality of the three other number input components, hence my desire to remove them :)

Reply:
https://github.com/openmsupply/open-msupply/issues/1440

Reply:
> #1440

Nice -- I thought I'd encountered this before. Have added follow-up comment there to say that this is half-done now: https://github.com/openmsupply/open-msupply/pull/2381#discussion_r1366006870